### PR TITLE
[apps] initial theming support

### PR DESCRIPTION
**Pull request by Simek**
# Why

Experiment introducing theming support in the workspace apps (Bare Expo, NCL and Test Suite).

# How

Utilize styleguide package, update Navigators, stack configs, add theme toggle, support cark mode in navigation components.

# Test Plan

The changes have been tested on Android and iOS hardware devices.

# Preview

### CODE REVIEW

**Review by tsapeta — COMMENTED** [review]
Looks great so far! 👏

File: apps/common/ThemeProvider.tsx
Comment (tsapeta) [inline apps/common/ThemeProvider.tsx:1]: Not sure if making a `common` directory here is a good idea (it's not an app). Can you temporarily copy these two files to both test-suite and NCL?

File: apps/common/ThemeToggler.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Ionicons } from '@expo/vector-icons';
+import { TouchableOpacity, Platform } from 'react-native';
+
+import { useTheme } from './ThemeProvider';
+
+export function ThemeToggle() {
```
Comment (tsapeta) [inline apps/common/ThemeToggler.tsx:6]: ThemeToggle or ThemeToggler (as the file name)? Maybe it should be a default export?

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: apps/common/ThemeProvider.tsx
Comment (Simek) [reply apps/common/ThemeProvider.tsx:1]: The extraction was needed to properly sync the themes between various navigators, is separated,, there was multiple different theme context, and toggling one did not affect other IIRC. Will retest this later today.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**